.github/workflows/codegen.yaml

```diff
@@ -8,17 +8,33 @@ name: codegen
 jobs:
   discovery:
     uses: googleapis/discovery-artifact-manager/.github/workflows/list-services.yml@master
-
-  generate:
+  batch:
     runs-on: ubuntu-20.04
     needs: discovery
+    outputs: 
+      batches: ${{ steps.chunk.outputs.result }}
+    steps:
+      - uses: actions/github-script@v5
+        id: chunk
+        with: 
+          script: |
+            console.log('splitting service names list into batches')
+            const services = ${{ needs.discovery.outputs.services }}
+            const MAX_BATCH_SIZE = 2
```

Comment:
Determine batch size as a function of the size of the service list (ensure max 256 batches)